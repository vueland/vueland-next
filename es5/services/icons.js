"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FaSizes = exports.MaterialSizes = exports.MaterialIcons = exports.FaIcons = void 0;
var FaIcons;
exports.FaIcons = FaIcons;

(function (FaIcons) {
  FaIcons["$add"] = "fas fa-plus";
  FaIcons["$delete"] = "fas fa-trash-alt";
  FaIcons["$sliders"] = "fas fa-sliders-h";
  FaIcons["$edit"] = "fas fa-pen";
  FaIcons["$menu"] = "fas fa-bars";
  FaIcons["$expand"] = "fas fa-chevron-down";
  FaIcons["$group"] = "fas fa-chevron-down";
  FaIcons["$subgroup"] = "fas fa-caret-down";
  FaIcons["$dropdown"] = "fas fa-caret-down";
  FaIcons["$radioOn"] = "far fa-dot-circle";
  FaIcons["$radioOff"] = "far fa-circle";
  FaIcons["$checkOn"] = "fas fa-check-square";
  FaIcons["$checkOff"] = "far fa-square";
  FaIcons["$arrowRight"] = "fas fa-chevron-right";
  FaIcons["$arrowLeft"] = "fas fa-chevron-left";
  FaIcons["$arrowUp"] = "fas fa-arrow-up";
  FaIcons["$filter"] = "fas fa-align-justify";
  FaIcons["$marker"] = "fas fa-map-marker-alt";
  FaIcons["$search"] = "fas fa-search";
  FaIcons["$close"] = "fas fa-times";
})(FaIcons || (exports.FaIcons = FaIcons = {}));

var MaterialIcons;
exports.MaterialIcons = MaterialIcons;

(function (MaterialIcons) {
  MaterialIcons["$add"] = "add";
  MaterialIcons["$delete"] = "delete_forever";
  MaterialIcons["$sliders"] = "fas fa-sliders-h";
  MaterialIcons["$edit"] = "fas fa-pen";
  MaterialIcons["$menu"] = "fas fa-bars";
  MaterialIcons["$expand"] = "fas fa-chevron-down";
  MaterialIcons["$group"] = "fas fa-chevron-down";
  MaterialIcons["$subgroup"] = "fas fa-caret-down";
  MaterialIcons["$dropdown"] = "fas fa-caret-down";
  MaterialIcons["$radioOn"] = "far fa-dot-circle";
  MaterialIcons["$radioOff"] = "far fa-circle";
  MaterialIcons["$checkOn"] = "check_box";
  MaterialIcons["$checkOff"] = "check_box_outline_blank";
  MaterialIcons["$arrowRight"] = "chevron_right";
  MaterialIcons["$arrowLeft"] = "chevron_left";
  MaterialIcons["$arrowUp"] = "arrow_upward";
  MaterialIcons["$filter"] = "filter_list";
  MaterialIcons["$marker"] = "fas fa-map-marker-alt";
  MaterialIcons["$search"] = "search";
  MaterialIcons["$close"] = "clear";
})(MaterialIcons || (exports.MaterialIcons = MaterialIcons = {}));

var MaterialSizes;
exports.MaterialSizes = MaterialSizes;

(function (MaterialSizes) {
  MaterialSizes[MaterialSizes["xs"] = 14] = "xs";
  MaterialSizes[MaterialSizes["s"] = 18] = "s";
  MaterialSizes[MaterialSizes["md"] = 20] = "md";
  MaterialSizes[MaterialSizes["l"] = 28] = "l";
  MaterialSizes[MaterialSizes["xl"] = 36] = "xl";
})(MaterialSizes || (exports.MaterialSizes = MaterialSizes = {}));

var FaSizes;
exports.FaSizes = FaSizes;

(function (FaSizes) {
  FaSizes[FaSizes["xs"] = 12] = "xs";
  FaSizes[FaSizes["s"] = 14] = "s";
  FaSizes[FaSizes["md"] = 18] = "md";
  FaSizes[FaSizes["l"] = 24] = "l";
  FaSizes[FaSizes["xl"] = 32] = "xl";
})(FaSizes || (exports.FaSizes = FaSizes = {}));
//# sourceMappingURL=icons.js.map