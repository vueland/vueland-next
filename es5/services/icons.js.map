{"version":3,"sources":["../../../src/services/icons.ts"],"names":[],"mappings":";;;;;;AAAA,IAAY,OAAZ;;;AAAA,CAAA,UAAY,OAAZ,EAAmB;AACjB,EAAA,OAAA,CAAA,OAAA,CAAA,GAAA,aAAA;AACA,EAAA,OAAA,CAAA,SAAA,CAAA,GAAA,qBAAA;AACA,EAAA,OAAA,CAAA,QAAA,CAAA,GAAA,qBAAA;AACA,EAAA,OAAA,CAAA,WAAA,CAAA,GAAA,mBAAA;AACA,EAAA,OAAA,CAAA,WAAA,CAAA,GAAA,mBAAA;AACA,EAAA,OAAA,CAAA,UAAA,CAAA,GAAA,mBAAA;AACA,EAAA,OAAA,CAAA,WAAA,CAAA,GAAA,eAAA;AACA,EAAA,OAAA,CAAA,UAAA,CAAA,GAAA,qBAAA;AACA,EAAA,OAAA,CAAA,WAAA,CAAA,GAAA,eAAA;AACA,EAAA,OAAA,CAAA,aAAA,CAAA,GAAA,sBAAA;AACA,EAAA,OAAA,CAAA,YAAA,CAAA,GAAA,qBAAA;AACD,CAZD,EAAY,OAAO,uBAAP,OAAO,GAAA,EAAA,CAAnB","sourcesContent":["export enum FaIcons {\n  $menu = 'fas fa-bars',\n  $expand = 'fas fa-chevron-down',\n  $group = 'fas fa-chevron-down',\n  $subgroup = 'fas fa-caret-down',\n  $dropdown = 'fas fa-caret-down',\n  $radioOn = 'far fa-dot-circle',\n  $radioOff = 'far fa-circle',\n  $checkOn = 'fas fa-check-square',\n  $checkOff = 'far fa-square',\n  $arrowRight = 'fas fa-chevron-right',\n  $arrowLeft = 'fas fa-chevron-left'\n}\n"],"sourceRoot":"","file":"icons.js"}