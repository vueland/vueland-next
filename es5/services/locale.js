"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.locale = void 0;
var locale = {
  ru: {
    months: ['Январь', 'Феввраль', 'Март', 'Апрель', 'Май', 'Июнь', 'Июль', 'Август', 'Сентябрь', 'Октябрь', 'Ноябрь', 'Декабрь'],
    monthsAbbr: ['Янв', 'Фев', 'Мар', 'Апр', 'Май', 'Июн', 'Июл', 'Авг', 'Сен', 'Окт', 'Ноя', 'Дек'],
    week: ['Вс', 'Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб']
  },
  en: {
    months: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
    monthsAbbr: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
    week: ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa']
  }
};
exports.locale = locale;
//# sourceMappingURL=locale.js.map