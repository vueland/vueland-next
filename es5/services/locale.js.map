{"version":3,"sources":["../../../src/services/locale.ts"],"names":[],"mappings":";;;;;;AAAO,IAAM,MAAM,GAAG;AACpB,EAAA,EAAE,EAAE;AACF,IAAA,MAAM,EAAE,CACN,KADM,EAEN,KAFM,EAGN,KAHM,EAIN,KAJM,EAKN,KALM,EAMN,KANM,EAON,KAPM,EAQN,KARM,EASN,KATM,EAUN,KAVM,EAWN,KAXM,EAYN,KAZM,CADN;AAeF,IAAA,IAAI,EAAE,CACJ,IADI,EAEJ,IAFI,EAGJ,IAHI,EAIJ,IAJI,EAKJ,IALI,EAMJ,IANI,EAOJ,IAPI;AAfJ,GADgB;AA0BpB,EAAA,EAAE,EAAE;AACF,IAAA,MAAM,EAAE,CACN,KADM,EAEN,KAFM,EAGN,KAHM,EAIN,KAJM,EAKN,KALM,EAMN,KANM,EAON,KAPM,EAQN,KARM,EASN,KATM,EAUN,KAVM,EAWN,KAXM,EAYN,KAZM,CADN;AAeF,IAAA,IAAI,EAAE,CACJ,IADI,EAEJ,IAFI,EAGJ,IAHI,EAIJ,IAJI,EAKJ,IALI,EAMJ,IANI,EAOJ,IAPI;AAfJ;AA1BgB,CAAf","sourcesContent":["export const locale = {\n  ru: {\n    months: [\n      'Янв',\n      'Фев',\n      'Мар',\n      'Апр',\n      'Май',\n      'Июн',\n      'Июл',\n      'Авг',\n      'Сен',\n      'Окт',\n      'Ноя',\n      'Дек',\n    ],\n    week: [\n      'Вс',\n      'Пн',\n      'Вт',\n      'Ср',\n      'Чт',\n      'Пт',\n      'Сб',\n    ],\n  },\n  en: {\n    months: [\n      'Jan',\n      'Feb',\n      'Mar',\n      'Apr',\n      'May',\n      'Jun',\n      'Jul',\n      'Aug',\n      'Sep',\n      'Oct',\n      'Nov',\n      'Dec',\n    ],\n    week: [\n      'Su',\n      'Mo',\n      'Tu',\n      'We',\n      'Th',\n      'Fr',\n      'Sa',\n    ],\n  },\n}\n"],"sourceRoot":"","file":"locale.js"}