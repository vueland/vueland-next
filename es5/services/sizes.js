"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sizes = void 0;
var Sizes;
exports.Sizes = Sizes;

(function (Sizes) {
  Sizes["xSmall"] = "12px";
  Sizes["small"] = "16px";
  Sizes["default"] = "24px";
  Sizes["medium"] = "28px";
  Sizes["large"] = "36px";
  Sizes["xLarge"] = "40px";
})(Sizes || (exports.Sizes = Sizes = {}));
//# sourceMappingURL=sizes.js.map