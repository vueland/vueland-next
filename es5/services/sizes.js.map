{"version":3,"sources":["../../../src/services/sizes.ts"],"names":[],"mappings":";;;;;;AAAA,IAAY,KAAZ;;;AAAA,CAAA,UAAY,KAAZ,EAAiB;AACf,EAAA,KAAA,CAAA,QAAA,CAAA,GAAA,MAAA;AACA,EAAA,KAAA,CAAA,OAAA,CAAA,GAAA,MAAA;AACA,EAAA,KAAA,CAAA,SAAA,CAAA,GAAA,MAAA;AACA,EAAA,KAAA,CAAA,QAAA,CAAA,GAAA,MAAA;AACA,EAAA,KAAA,CAAA,OAAA,CAAA,GAAA,MAAA;AACA,EAAA,KAAA,CAAA,QAAA,CAAA,GAAA,MAAA;AACD,CAPD,EAAY,KAAK,qBAAL,KAAK,GAAA,EAAA,CAAjB","sourcesContent":["export enum Sizes {\n  xSmall = \"12px\",\n  small = \"16px\",\n  default = \"24px\",\n  medium = \"28px\",\n  large = \"36px\",\n  xLarge = \"40px\",\n}\n"],"sourceRoot":"","file":"sizes.js"}