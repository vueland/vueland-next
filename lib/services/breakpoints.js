export const breakpoints = {
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200,
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYnJlYWtwb2ludHMuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi8uLi9wYWNrYWdlcy92dWVsYW5kL3NyYy9zZXJ2aWNlcy9icmVha3BvaW50cy50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxNQUFNLENBQUMsTUFBTSxXQUFXLEdBQUc7SUFDekIsRUFBRSxFQUFFLEdBQUc7SUFDUCxFQUFFLEVBQUUsR0FBRztJQUNQLEVBQUUsRUFBRSxHQUFHO0lBQ1AsRUFBRSxFQUFFLElBQUk7Q0FDVCxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0IGNvbnN0IGJyZWFrcG9pbnRzID0ge1xuICBzbTogNTc2LFxuICBtZDogNzY4LFxuICBsZzogOTkyLFxuICB4bDogMTIwMCxcbn1cbiJdfQ==