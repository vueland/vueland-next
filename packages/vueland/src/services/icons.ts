export enum FaIcons {
  $add = 'fas fa-plus',
  $delete = 'fas fa-trash-alt',
  $sliders = 'fas fa-sliders-h',
  $edit = 'fas fa-pen',
  $menu = 'fas fa-bars',
  $chevronDown = 'fas fa-chevron-down',
  $caretDown = 'fas fa-caret-down',
  $radioOn = 'far fa-dot-circle',
  $radioOff = 'far fa-circle',
  $box = 'far fa-square',
  $checkbox = 'far fa-check-square',
  $arrowRight = 'fas fa-chevron-right',
  $arrowLeft = 'fas fa-chevron-left',
  $arrowUp = 'fas fa-arrow-up',
  $filter = 'fas fa-align-justify',
  $marker = 'fas fa-map-marker-alt',
  $search = 'fas fa-search',
  $close = 'fas fa-times',
  $paperclip = 'fas fa-paperclip'
}

export enum MaterialIcons {
  $add = 'add',
  $delete = 'delete_forever',
  $chevronDown = 'expand_more',
  $group = 'expand_more',
  $caretDown = 'arrow_drop_down',
  $box = 'check_box',
  $checkbox = 'check_box_outline_blank',
  $arrowRight = 'chevron_right',
  $arrowLeft = 'chevron_left',
  $arrowUp = 'arrow_upward',
  $filter = 'filter_list',
  $search = 'search',
  $close = 'clear',
}
