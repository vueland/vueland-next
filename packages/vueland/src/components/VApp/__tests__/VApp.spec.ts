import { mount } from '@vue/test-utils'
import { defineComponent, inject } from 'vue'
import { VApp } from '../index'
import { breakpoints } from '../../../services/breakpoints'
import 'regenerator-runtime'

const THROTTLING_TIMEOUT = 20

describe('VApp', () => {
  let bp

  const global = {
    innerWidth: breakpoints.sm,
  }

  const onResize = (width) => {
    global.innerWidth = width
  }

  const mountComponent = (options = { props: { global } }) => mount(VApp, {
    ...options,
    slots: {
      default: defineComponent({
        setup() {
          bp = inject('$v_breakpoints')
        },
        template: `
          <div></div>`,
      }),
    },
  })

  it('should mount component and match snapshot', () => {
    const wrapper = mountComponent()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should test "sm" size', () => {
    mountComponent()

    expect(bp.current).toEqual('sm')
    expect(bp.smAndLess).toBe(true)
  })

  it('should test "md" size', () => {
    global.innerWidth = breakpoints.md
    mountComponent()

    expect(bp.current).toEqual('md')
    expect(bp.mdAndLess).toBe(true)
  })

  it('should test "lg" size', () => {
    global.innerWidth = breakpoints.lg
    mountComponent()

    expect(bp.current).toEqual('lg')
    expect(bp.lgAndLess).toBe(true)
  })

  it('should test "xl" size', () => {
    global.innerWidth = breakpoints.xl
    mountComponent()

    expect(bp.current).toEqual('xl')
    expect(bp.xlAndLess).toBe(true)
  })

  it('should test resize listener', async () => {
    mountComponent()
    const event = new Event('resize')
    let arg = breakpoints.lg

    window.addEventListener('resize', () => onResize(arg))
    window.dispatchEvent(event)

    await new Promise(res => setTimeout(res, THROTTLING_TIMEOUT, true))

    expect(bp.current).toEqual('lg')

    arg = breakpoints.md

    window.addEventListener('resize', () => onResize(arg))
    window.dispatchEvent(event)

    await new Promise(res => setTimeout(res, THROTTLING_TIMEOUT, true))

    expect(bp.current).toEqual('md')

    arg = breakpoints.sm

    window.addEventListener('resize', () => onResize(arg))
    window.dispatchEvent(event)

    await new Promise(res => setTimeout(res, THROTTLING_TIMEOUT, true))

    expect(bp.current).toEqual('sm')
  })
})
