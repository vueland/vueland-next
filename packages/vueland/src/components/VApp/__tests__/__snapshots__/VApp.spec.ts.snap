// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VApp should mount component and match snapshot 1`] = `
<div class="v-app">
  <div class="v-app--wrapper">
    <div>
    </div>
  </div>
</div>
`;
